File: SmartDeliverySystem/Models/Delivery.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SmartDeliverySystem.Models
{
    public enum DeliveryStatus
    {
        PendingPayment,
        Paid,
        Assigned,
        InTransit,
        Delivered,
        Cancelled
    }

    public class Delivery
    {
        [Key]
        public int Id { get; set; }

        [ForeignKey("Vendor")]
        public int VendorId { get; set; }

        [ForeignKey("Store")]
        public int StoreId { get; set; }

        public decimal TotalAmount { get; set; }
        public DeliveryStatus Status { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? PaymentDate { get; set; }
        public string? PaymentMethod { get; set; }
        public decimal? PaidAmount { get; set; }
        public DateTime? AssignedAt { get; set; }
        public string? DriverId { get; set; }
        public string? GpsTrackerId { get; set; }

        // Location coordinates
        public double? FromLatitude { get; set; }
        public double? FromLongitude { get; set; }
        public double? ToLatitude { get; set; }
        public double? ToLongitude { get; set; }
        public double? CurrentLatitude { get; set; }
        public double? CurrentLongitude { get; set; }
        public DateTime? LastLocationUpdate { get; set; }
        public DateTime? DeliveredAt { get; set; }
        public string? TrackingNotes { get; set; }

        // Navigation properties
        public virtual Vendor Vendor { get; set; } = null!;
        public virtual Store Store { get; set; } = null!;
        public virtual ICollection<DeliveryProduct> Products { get; set; } = [];
        public virtual ICollection<DeliveryLocationHistory> LocationHistory { get; set; } = [];
    }

    // New GPS tracking model
    public class DeliveryLocationHistory
    {
        public int Id { get; set; }
        public int DeliveryId { get; set; }
        public Delivery? Delivery { get; set; }
        public double Latitude { get; set; }
        public double Longitude { get; set; }
        public DateTime Timestamp { get; set; } = DateTime.UtcNow;
        public string? Notes { get; set; }
        public double? Speed { get; set; } // km/h
    }
}

File: SmartDeliverySystem.Frontend/src/types/delivery.ts
export interface LocationData {
    id?: number;
    name: string;
    latitude: number;
    longitude: number;
}

export interface ProductData {
    id?: number;
    name: string;
    weight: number;
    category: string;
    price: number;
    vendorId?: number;
}

export interface VendorData extends LocationData {
    products?: ProductData[];
}

export interface StoreData extends LocationData {
    inventory?: InventoryItem[];
}

export interface ProductData {
    id: number;
    name: string;
    weight: number;
    category: string;
    price: number;
    vendorId: number;
}

export interface InventoryItem {
    id?: number;
    productId: number;
    productName: string;
    category: string;
    quantity: number;
    price: number;
    storeId: number;
}

export interface DeliveryData {
    deliveryId: number;
    vendorId: number;
    storeId: number;
    status: DeliveryStatus;
    currentLatitude?: number;
    currentLongitude?: number;
    vendorLatitude?: number;
    vendorLongitude?: number;
    storeLatitude?: number;
    storeLongitude?: number;
    fromLatitude?: number;
    fromLongitude?: number;
    toLatitude?: number;
    toLongitude?: number;
    driverId?: string;
    gpsTrackerId?: string;
    lastLocationUpdate?: string;
    totalAmount?: number;
    products?: DeliveryProductData[];
}

export interface DeliveryProductData {
    productId: number;
    quantity: number;
    price?: number;
}

export interface DeliveryRequest {
    vendorId: number;
    storeId: number;
    products: DeliveryProductData[];
}

export interface PaymentData {
    amount: number;
    paymentMethod: 'CreditCard' | 'Cash' | 'BankTransfer';
}

export interface DriverData {
    driverId: string;
    gpsTrackerId: string;
}

export interface LocationUpdateData {
    deliveryId: number;
    latitude: number;
    longitude: number;
    timestamp: string;
    speed?: number;
    notes?: string;
}

export type DeliveryStatus =
    | 'Pending'
    | 'PendingPayment'
    | 'Assigned'
    | 'InTransit'
    | 'Delivered'
    | 'Paid'
    | 'Cancelled';

export type ConnectionStatus = 'connected' | 'disconnected' | 'error';
